### DISCUSSION TITLE
How to test HttpClient inside API Tests

### DISCUSSION THREAD

**Discussion by giscus** [linkdotnet/Blog.Discussions#115] [General]
# How to test HttpClient inside API Tests

I am a great fan of API tests. I do like to utilize the WebApplicationFactory for this. I even wrote some articles about this in the past "Introduction to WebApplicationFactory". But what if we need to "mock" HttpClient calls inside our tests?

https://steven-giesel.com/blogPost/47e4a351-0f1b-4bb5-a3d5-0ce4d7c1fa61/how-to-test-httpclient-inside-api-tests

**Comment by diegodrf** [comment]
Cool!
I use the [MockHttp package](https://www.nuget.org/packages/RichardSzalay.MockHttp) to mock my API tests, but it's good to know how easy it is to do without package.

**Reply by linkdotnet** [reply]
Oh nice - I would assume they do the same thing under the hood.
They just have a very convenient builder pattern!
